.github/workflows/algolia-upload-index.yml

```diff
@@ -16,8 +16,15 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v2
         with:
-          node-version: "12"
+          node-version: 16
+          cache: yarn
 
-      # Runs a single command using the runners shell
-      - name: Upload to algolia
-        run: cd .github/workflows/algolia && sh algolia_upload_script.sh
+      - name: Reduce the number of dependencies to install by hiding workspaces
+        run: |
+          npx replace '"workspaces":' '"ignored-workspaces":' package.json
```

Comment:
This trick should reduce `yarn install` time by skipping all dependencies in all workspaces. We might be able to do something more ‘official’ after upgrading to Yarn Berry — docs: https://yarnpkg.com/cli/workspaces/focus

Focus-mode does not work for root package.json in Yarn 1 AFAIU.